cmd/api/src/api/v2/saved_queries.go

```diff
@@ -100,3 +104,26 @@ func (s Resources) CreateSavedQuery(response http.ResponseWriter, request *http.
 		api.WriteBasicResponse(request.Context(), savedQuery, http.StatusCreated, response)
 	}
 }
+
+func (s Resources) DeleteSavedQuery(response http.ResponseWriter, request *http.Request) {
+	var (
+		rawSavedQueryID = mux.Vars(request)[api.URIPathVariableSavedQueryID]
+		userID          = ctx2.FromRequest(request).AuthCtx.Session.UserID
+	)
+
+	if savedQueryID, err := strconv.Atoi(rawSavedQueryID); err != nil {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, api.ErrorResponseDetailsIDMalformed, request), response)
+	} else if savedQueryBelongsToUser, err := s.DB.SavedQueryBelongsToUser(userID, savedQueryID); err != nil {
```

Comment:
In most cases, we won't hit the `database.ErrNotFound` in the delete call, we'll hit it here first since if the id was deleted, this query will fail. We need to check for `database.ErrNotFound` and handle it in this section of code. We could optionally handle it below as well just for thoroughness, but it would be an edge case. You can chain multiple  error checks without nesting as well:

```suggestion
	} else if savedQueryBelongsToUser, err := s.DB.SavedQueryBelongsToUser(userID, savedQueryID); errors.Is(err, database.ErrNotFound) {
	// 404
	} else if err != nil {
	// 500
	} else if !savedQueryBelongsToUser {
	//400
	} else if err := s.DB.DeleteSavedQuery; errors.Is(err, database.ErrNotFound) {
	// 404
	} else if err != nil {
	// 500
	} else {
	// 204
	}
```